puppet/modules/jenkins_job_builder/files/theforeman.org/scripts/test/test_develop.sh

```diff
@@ -24,6 +24,15 @@ bundle install --without=development --jobs=5 --retry=5
 # we need to install node modules for integration tests (which only run on postgresql)
 if [ ${database} = postgresql -a -e "$APP_ROOT/package.json" ]; then
   npm install npm@'<6.0.0' # first upgrade to newer npm
+
+  # Test asset precompile
+  NODE_ENV=production $APP_ROOT/node_modules/.bin/npm install
```

Comment:
I am not sure I understand this line,
I never seen projects running npm from `./node_modules/.bin/npm`.

Does it mean we are running a local `npm` that we install using the global `npm`?
Why not to use the global `npm`?

Is this the line that install the `npm` locally with `npm`? `npm install npm@'<6.0.0'`?
Can we change it to `npm install --global npm@'<6.0.0'`?

Sorry for all the questions, I just want to make sure I understand.

Reply:
We generally don't want to use any global because it can interfere with other runs. The same jenkins slave could run some other job in parallel that installs another npm version. This provides us with some isolation. webpack-rails [does the same](https://github.com/mipearson/webpack-rails/blob/83a5d4c8af82e1f33116e3550d3bc3e8ecff977f/lib/webpack/railtie.rb#L18).